src/Worker.Extensions.DurableTask/DefaultDurableClientContext.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+using DurableTask;
+using DurableTask.Grpc;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Http;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+/// <summary>
+/// Concreate implementation of the <see cref="DurableClientContext"/> abstract class that
+/// allows callers to start and manage orchestration instances.
+/// </summary>
+internal sealed class DefaultDurableClientContext : DurableClientContext
+{
+    private readonly DurableClientInputData inputData;
+
+    // Private constructor that's called by the converter inner-class
+    private DefaultDurableClientContext(DurableTaskClient client, DurableClientInputData inputData)
+    {
+        this.Client = client ?? throw new ArgumentNullException(nameof(client));
+        this.inputData = inputData ?? throw new ArgumentNullException(nameof(inputData));
+
+        ArgumentNullException.ThrowIfNull(inputData.taskHubName, nameof(inputData.taskHubName));
+        ArgumentNullException.ThrowIfNull(inputData.requiredQueryStringParameters, nameof(inputData.requiredQueryStringParameters));
+    }
+
+    /// <inheritdoc/>
+    public override DurableTaskClient Client { get; }
+
+    /// <inheritdoc/>
+    public override string TaskHubName => this.inputData.taskHubName;
+
+    /// <inheritdoc/>
+    public override HttpResponseData CreateCheckStatusResponse(HttpRequestData request, string instanceId, bool returnInternalServerErrorOnFailure = false)
+    {
+        // TODO: To better support scenarios involving proxies or application gateways, this
+        //       code should take the X-Forwarded-Host, X-Forwarded-Proto, and Forwarded HTTP
+        //       request headers into consideration and generate the base URL accordingly.
+        //       More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
+        //       One potential workaround is to set ASPNETCORE_FORWARDEDHEADERS_ENABLED to true.
```

Comment:
Any reason why we don't already use `ASPNETCORE_FORWARDEDHEADERS_ENABLED`?

Reply:
I'm not sure. I think Connor was the one who looked into this originally. I'm just mentioning this here since it was mentioned in another GitHub issue. I don't currently know if it works or not.